### PR TITLE
Testing the conditions for the special checks.

### PR DESCRIPTION
**Pull request by felipemontoya**
# Description

This PR adds testing to the null checks and makes the code conform to this tests

# Reviewers

@juancamilom, could you please look at the test cases and let me know if I have something wrong there?
I have only made one of the functions conform, but I will come back to this later on.

### CODE REVIEW

**Comment by juancamilom** [comment]
@felipemontoya 
I think your logic is fine for that second case (has_null). 
The first case (are_all_not_null)  requires true when the list is not empty AND all_problems_were answered.
The third case (are_all_null)  requires true when:
not problems_to_answer or not_answered_quantity == len(problems_to_answer)

File: flow_control/tests/test_flowcontrol.py [hunk 1]
```diff
@@ -185,3 +185,58 @@ def test_author_view(self, fragment_mock):
         fragment_mock.assert_called()
         fragment_instance.add_javascript.assert_called()
         fragment_instance.initialize_js.assert_called()
+
```
Comment (felipemontoya) [inline flow_control/tests/test_flowcontrol.py:188]: @juancamilom do you think this assertions correctly show what should be happening?

File: flow_control/tests/test_flowcontrol.py [hunk 2]
```diff
@@ -185,3 +185,58 @@ def test_author_view(self, fragment_mock):
...
+        """
+        Test the conditions of which it passes the are_all_null check
+        Description: none of the problems have been answered
+        """
+        self.assertFalse(self.block.are_all_null([1.0]))
+        self.assertFalse(self.block.are_all_null([1.0, 1.0]))
+        self.assertFalse(self.block.are_all_null([1.0, 1.0, 1.0]))
+        self.assertFalse(self.block.are_all_null([1.0, 0.0]))
+        self.assertFalse(self.block.are_all_null([1.0, None]))
+        self.assertFalse(self.block.are_all_null([1.0, None]))
```
Comment (juancamilom) [inline flow_control/tests/test_flowcontrol.py]: this one is duplicated

File: flow_control/tests/test_flowcontrol.py [hunk 3]
```diff
@@ -185,3 +185,58 @@ def test_author_view(self, fragment_mock):
...
+
+        self.assertTrue(self.block.are_all_null([None, None]))
+        self.assertTrue(self.block.are_all_null([]))
+        self.assertTrue(self.block.are_all_null([0.0]))
+        self.assertTrue(self.block.are_all_null([None]))
+        self.assertTrue(self.block.are_all_null([0.0, None]))
+
```
Comment (juancamilom) [inline flow_control/tests/test_flowcontrol.py]: +        self.assertTrue(self.block.are_all_null([0.0, 0.0]))

File: flow_control/tests/test_flowcontrol.py [hunk 4]
```diff
@@ -185,3 +185,58 @@ def test_author_view(self, fragment_mock):
...
+    def test_are_all_not_null(self):
+        """
+        Test the conditions of which it passes the are_all_not_null check
+        Description: all problems have been answered
+        """
+        self.assertTrue(self.block.are_all_not_null([1.0]))
+        self.assertTrue(self.block.are_all_not_null([1.0, 1.0]))
+        self.assertTrue(self.block.are_all_not_null([1.0, 1.0, 1.0]))
+        self.assertTrue(self.block.are_all_not_null([1.0, 0.0]))
+        self.assertTrue(self.block.are_all_not_null([1.0, None]))
```
Comment (juancamilom) [inline flow_control/tests/test_flowcontrol.py]: duplicated and wrong (I believe)
+        self.assertFalse(self.block.are_all_not_null([1.0, None]))

File: flow_control/tests/test_flowcontrol.py (hunk 1 above)
Comment (juancamilom) [reply flow_control/tests/test_flowcontrol.py]: @felipemontoya I've added 3 inline comments

**Comment by felipemontoya** [comment]
@juancamilom I think this now passes the right tests and it is ready to go

**Pull request merged**